pkg/motor/motor.go

```diff
@@ -94,14 +97,40 @@ func initMotor(mtr *motorNodeImpl, options ...mpc.WalletOption) (err error) {
 		return err
 	}
 	mtr.DID = *baseDid
+	log.Println("Wallet set to:", mtr.Address)
+	return nil
+}
 
-	// It creates a new host.
-	mtr.SonrHost, err = host.NewDefaultHost(context.Background(), config.DefaultConfig(config.Role_MOTOR))
-	if err != nil {
-		return err
+func (mtr *motorNodeImpl) Connect() error {
+	if mtr.Wallet == nil {
+		return fmt.Errorf("wallet is not initialized")
+	}
+
+	if mtr.SonrHost != nil {
+		log.Println("Host already connected")
+		return nil
+	}
+
+	var err error
+	// Create new host
+	if mtr.isHostEnabled {
+		log.Println("Creating host...")
+		mtr.SonrHost, err = host.NewDefaultHost(context.Background(), config.DefaultConfig(config.Role_MOTOR, mtr.Address))
+		if err != nil {
+			return err
+		}
+	} else {
+		return fmt.Errorf("host is not enabled")
 	}
 
-	// Create motorNodeImpl
+	// Utilize discovery protocol
+	if mtr.isDiscoveryEnabled {
+		log.Println("Enabling Discovery...")
+		mtr.discovery, err = dp.New(context.Background(), mtr.SonrHost, mtr.callback)
```

Comment:
is the call to discover peers explicit? motor stand up might take a long time if we are doing it during initialization.

Reply:
Yeah this is why its done when you call `Connect()` method from bind. Allowing MPC generation prior to bootstrapping @joshLong145